### PR TITLE
Allows core shapes to be overridden

### LINKED ISSUE #2415: Cannot override base shapes in custom Theme.
I created a Message.cshtml file in my theme to style differently the message shape . But it  is  not picked up from my theme. It still uses the default shape coming from  :

OrchardCore.DisplayManagement.Shapes.CoreShapes.Message method.

This was working before I upgraded to the latest  version in dev branch.

### PR DESCRIPTION
**Pull request by jtkech**
Fixes #2415.

### CODE REVIEW

**Comment by deanmarcussen** [comment]
@jtkech You know me, always super curious to see how all the magic gets plugged together ;)

So I am also super curious about the fix. 

Have you got time for a quick brain dump again? 
And I'll see if I can get it into the docs somewhere.

So this means that `OrchardcCore.Cms.Web` always wins if it has an overridden shape or view template? (Which sounds pretty cool.)

Does this mean that if, for example, you want to override a view for `TheAdmin`, you can do it at the startup project level, instead of having to create a custom admin theme, and that will then affect all tenants running `TheAdmin` theme?

And maybe then you can also override a view for all themes by putting a template in the startup project?

Which also sounds super useful.

**Comment by sebastienros** [comment]
@deanmarcussen  I think that was the intent with making the app behave as a master module.

File: src/OrchardCore.Modules/OrchardCore.Contents/Startup.cs [hunk 1]
```diff
@@ -42,6 +44,13 @@ public class Startup : StartupBase
     {
         public override void ConfigureServices(IServiceCollection services)
         {
+            // Re-register core shapes to not be tied to the application's module,
+            // so that that they can be overridden by the current theme.
+            services.ReplaceShapeAttributes<CoreShapes>();
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Contents/Startup.cs]: This is not a valid solution. These shapes are part of the DisplayManagement project, which can be used with Contents. Think about a SaaS solution for instance. The solution would be to add them to a fake feature, or a display management feature, that would be low priority. Even if it means to change the `AddShapeAttribute` to accept an extra parameter to mention it need to use a default feature. Note that this feature could be a static variable one, like `FeatureInfo.Default` that could be used when necessary.

Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Contents/Startup.cs]: Yes, the idea was that if `OC.Contents` is not enabled, these core shapes are still registered through `DisplayManagement`.

> The solution would be to add them to a fake feature

I thought about this but it needs more code, e.g a type can be tied to a feature if it is registered through a startup. Through the application we already have helpers to do pseudo startups but there are all tied to the application's module, we don't have any helper yet that would tie a given pseudo startup to a fake feature.

But i could think about it if it is worth to work on it, let me know.

**Comment by jtkech** [comment]
@deanmarcussen yes, exactly what you said.

**Comment by jtkech** [comment]
Okay, i found a way to create an always enabled core feature on which we can tie a given type just by using the orchard `FeatureAttribute`, so that we will be able to use it for other things than shapes.

Beautiful ;)

Need to do some tweaks then i will update this PR soon

**Comment by sebastienros** [comment]
You see, I just need to push you a little bit, and you always find a simpler, more natural way.

**Comment by sebastienros** [comment]
I really don't like "Secondary" I prefer to use "Default" like it was and "ApplicationFeature" or "MainFeature" for the application one.

**Comment by jtkech** [comment]
Okay, i just changed the names just to show the default / secondary one after the main / primary one in the admin features page ;)

Otherwise, as you i prefer Main and Default, i will update it.

**Pull request merged**